api/src/main/java/org/eclipse/microprofile/jwt/builder/Jwt.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.builder;
+
+import java.util.Map;
+
+import org.eclipse.microprofile.jwt.builder.spi.JwtProvider;
+
+/**
+ * Factory class for creating {@link JwtClaimsBuilder} which produces
+ * signed, encrypted or signed first and then encrypted JWT tokens.
+ *
+ * <p>
+ * The following example shows how to initialize a {@link JwtClaimsBuilder} from an existing resource
+ * containing the claims in a JSON format and produce a signed JWT token with a configured signing key:
+ * 
+ * <pre>
+ * <code>
+ * String = Jwt.claims("/tokenClaims.json").sign();
+ * </code>
+ * </pre>
+ * <p>
+ * The next example shows how to use {@link JwtClaimsBuilder} to add the claims and encrypt a JSON
+ * representation of these claims with a configured encrypting key:
+ * 
+ * <pre>
+ * <code>
+ * String = Jwt.claims().issuer("https://issuer.org").claim("custom-claim", "custom-value").encrypt();
```

Comment:
encryption is not supported by MP JWT spec yet, so it this already useful?

Reply:
It will be supported, see #58. So we need to expect it, and once again, MP JWT should be able to reach beyond MP JWT servers, same as we want MP JWT servers to accept as many tokens as possible